// @codedoc/coding-blog-plugin/dist/es5/components/quote.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Quote = exports.QuoteStyle = void 0;
var jss_theme_1 = require("@connectv/jss-theme");
exports.QuoteStyle = jss_theme_1.themedStyle(function (theme) { return ({
    quote: {
        fontSize: 20,
        padding: '0 56px',
    },
    text: {
        position: 'relative',
        padding: '1px 0',
        '& .begin, & .end': {
            fontSize: 48,
            position: 'absolute',
            opacity: .25,
            '&.begin': {
                top: 8, left: -56,
                transform: 'rotate(180deg)'
            },
            '&.end': {
                bottom: 8, right: -56,
            }
        },
    },
    source: {
        textAlign: 'right',
        fontStyle: 'italic',
    }
}); });
function Quote(options, renderer, content) {
    var classes = this.theme.classes(exports.QuoteStyle);
    return renderer.create("div", { class: classes.quote },
        renderer.create("div", { class: classes.text },
            renderer.create("div", { class: "icon-font begin" }, "format_quote"),
            content,
            renderer.create("div", { class: "icon-font end" }, "format_quote")),
        options.from ? renderer.create("div", { class: classes.source }, options.from) : '');
}
exports.Quote = Quote;
