// @codedoc/coding-blog-plugin/dist/es6/components/quote.js
import { themedStyle } from '@connectv/jss-theme';
export const QuoteStyle = themedStyle(theme => ({
    quote: {
        fontSize: 20,
        padding: '0 56px',
    },
    text: {
        position: 'relative',
        padding: '1px 0',
        '& .begin, & .end': {
            fontSize: 48,
            position: 'absolute',
            opacity: .25,
            '&.begin': {
                top: 8, left: -56,
                transform: 'rotate(180deg)'
            },
            '&.end': {
                bottom: 8, right: -56,
            }
        },
    },
    source: {
        textAlign: 'right',
        fontStyle: 'italic',
    }
}));
export function Quote(options, renderer, content) {
    const classes = this.theme.classes(QuoteStyle);
    return renderer.create("div", { class: classes.quote },
        renderer.create("div", { class: classes.text },
            renderer.create("div", { class: "icon-font begin" }, "format_quote"),
            content,
            renderer.create("div", { class: "icon-font end" }, "format_quote")),
        options.from ? renderer.create("div", { class: classes.source }, options.from) : '');
}
